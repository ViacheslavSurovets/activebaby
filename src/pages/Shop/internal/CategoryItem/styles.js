import styled, { css } from 'styled-components';
import { tablet, mobile } from '@media';


const styleCategoryItemBig = css`
  width: 40vw;
`;

const styleCategoryItemSmall = css`
  width: 30vw;
`;

const getSize = ( props ) => {
  if ( props.size === 'big' ) {
    return styleCategoryItemBig;
  }
  if ( props.size === 'small' ) {
    return styleCategoryItemSmall;
  }

  return;
};

const styleItemTransformOnHover = css`
  transform: scale(1.03);
  transition: all 1s ease;
  box-shadow: 4px 4px 4px -3px ${ ( { theme } ) => theme.black };
`;

export const CategoryItemContainer = styled.div`
  display: flex;
  flex: 1 1 auto;
  height: 30vw;
  position: relative;
  margin: 1rem;
  transition: all 1s;
  ${ getSize };
  ${ tablet },
  ${ mobile }{
    width: 90vw;
    height: 70vw;
  }
  .js-categoryItemImage:hover {
    ${ styleItemTransformOnHover };
  }
`;

export const CustomButtonWrapper = styled.div`
  display: flex;
  position: absolute;
  bottom: 10%;
  width: 50%;
  left: 50%;
  margin-left: -25%;
  z-index: 1;
  &:hover{
    ~ .js-categoryItemImage {
      ${ styleItemTransformOnHover };
    }
  }
`;

export const CategoryItemImg = styled.div`
  width: 100%;
  height: 100%; 
  outline: none;
  transition: all 1s;
  background-image: url(${ props => props.background });
  background-size: cover;
  background-position:unset;
`;


// export const CategoryItemTitle = styled.span`
//   ${commonStyleTextFourRem};
//   ${mobile}{
//     ${commonStyleTextThreeRem};
//   }
// `;




