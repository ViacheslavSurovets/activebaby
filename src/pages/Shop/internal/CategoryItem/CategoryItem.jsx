import React from 'react';
import { useTranslation } from 'react-i18next';
import { withRouter } from 'react-router-dom';
import { CustomButton } from '@components';
import { CategoryItemContainer, CustomButtonWrapper, CategoryItemImg } from './styles';
import PropTypes from 'prop-types';


const CategoryItem = ( { title, imageUrl, size, history, linkUrl, match } ) => {
  const { t, i18n } = useTranslation ();
  return (
    <CategoryItemContainer size={ size }>
      <CustomButtonWrapper>
        <CustomButton onClick={ () => history.push ( linkUrl ) } width={ '100%' }
                      fontSize={ 2 }>{ t ( title ) }</CustomButton>
      </CustomButtonWrapper>
      <CategoryItemImg className='js-categoryItemImage' background={ imageUrl } />
    </CategoryItemContainer>
  );
};

export default withRouter ( CategoryItem );

CategoryItem.propTypes = {
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  linkUrl: PropTypes.string.isRequired,
  match: PropTypes.object,
  history: PropTypes.any
};

