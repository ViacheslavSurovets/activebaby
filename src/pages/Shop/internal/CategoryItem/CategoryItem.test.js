import React from 'react';
import { shallow } from 'enzyme';
import CategoryItem from './CategoryItem';


describe ( 'CategoryItem', () => {

  it ( 'CategoryItem Snapshot', () => {
    expect ( shallow ( <CategoryItem  /> ) ).toMatchSnapshot ();
  } );
} );
