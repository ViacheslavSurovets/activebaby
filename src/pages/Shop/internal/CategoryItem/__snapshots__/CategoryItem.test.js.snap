// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem CategoryItem Snapshot 1`] = `ShallowWrapper {}`;
